### ISSUE TITLE
URL attribute for nodes

### ISSUE THREAD

**Issue by plion** [rich-iannone/DiagrammeR#243]
Hi,

I use your package for a year now to create graph in shiny apps. Users are are enthusiastic and ask me for enhancements like beeing able to click on a node to access specific url.

By looking at the graphviz doc and the generate_dot() function I found that adding **"URL"** to the **node_attributes** vector would solve my problem. I've done it and it worked.

 I'm not used enough to package developpement and to your package in particular to know if this is the right thing to do. I've seen edgeURL, headURL and tailURL already existed and was wondering why **URL** did not.

I'd like to know if this enhancement could be incorporated in your package and if yes how could I help ? (other tests, pull request, ...)

Thanks

**Comment by rich-iannone** [comment]
@plion Sorry for the delay in getting back to you... I got a chance to try out this parameter and indeed it works great! I modified the package a bit in the development version to allow creation of this attribute (I think I just missed it before).

FYI I'm also starting to develop helper functions (`node_aes()` and `edge_aes()`) that make it a bit easier to insert attribute values for nodes and edges. There are a few benefits here like documentation for these parameters, autocomplete in RStudio, and a clean separation between node and edge attributes (some of the attributes are named the same for nodes and edges!).

Right now, in the development version, only the `add_path()` function currently has support for this in the similarly named `node_aes` and `edge_aes` arguments. Try it out and let me know if this is useful or if there can be improvements!
